kernel/classes/datatypes/ezuser/ezuser.php

```diff
@@ -1415,16 +1415,18 @@ private static function generateUserCacheData( $userId )
                                         'password_hash'      => $user->attribute( 'password_hash' ),
                                         'password_hash_type' => $user->attribute( 'password_hash_type' ) );
 
+        // setting caching enabled flag to false is a protection against infinite recursion.
```

Comment:
please expand with something like explanation @peterkeung made:
`function groups relies on caching. The function generateUserCacheData relies on function groups. To avoid an infinitive loop, the code is disabling caching in function generateUserCacheData.`